### PR TITLE
Use classes and methods, various cleanup

### PR DESCRIPTION
**Pull request by vabold**

### CODE REVIEW

File: include/sdk/expHeap.hh [hunk 1]
```diff
@@ -31,23 +55,38 @@ struct MEMiExpBlockHead
...
 
-struct MEMiExpHeapHead : MEMiHeapHead
+class MEMiExpHeapHead : public MEMiHeapHead
 {
-    MEMiExpBlockList freeBlocks;
-    MEMiExpBlockList usedBlocks;
-    u16 groupId;
-    u16 attribute;
-};
+private:
```
Comment (malleoz) [inline include/sdk/expHeap.hh:65]: Is the `private` -> `public` -> `private` intentional? I kinda like this... This way we see the ctor and dtor first as we would typically expect, yet we can see the change in the access specifier.

File: include/sdk/heapCommon.hh [hunk 2]
```diff
@@ -2,58 +2,53 @@
...
+class MEMiHeapHead
 {
-    u32 signature;
-    MEMLink link;
-    MEMList childList;
-    void *heapStart;
-    void *heapEnd;
-    u16 optFlag;
-};
+protected:
```
Comment (malleoz) [inline include/sdk/heapCommon.hh:13]: In my opinion, I could get behind keeping the constructors always at the top regardless of access specifiers, but everything else (all the `fill` functions) should be moved to a new `protected` section below the `public` section.

File: src/sdk/mem_expHeap.cc [hunk 3]
```diff
@@ -1,413 +1,417 @@
...
+            reinterpret_cast<MEMiExpBlockHead *>( SubOffset( block, sizeof( MEMiExpBlockHead ) ) );
 
-    AppendMBlock_( &heap->usedBlocks, head );
+    Region region = head->getRegion( );
+    mUsedBlocks.remove( head );
+    recycleRegion( region );
+}
 
-    return address;
+u32 MEMiExpHeapHead::getAllocatableSize( s32 align )
```
Comment (malleoz) [inline src/sdk/mem_expHeap.cc]:
```suggestion
u32 MEMiExpHeapHead::getAllocatableSize( s32 align ) const
```

**Review by malleoz — CHANGES_REQUESTED** [review]

File: include/sdk/expHeap.hh (hunk 1 above)
Comment (vabold) [reply include/sdk/expHeap.hh:65]: Yes, it is.

File: include/sdk/heapCommon.hh (hunk 2 above)
Comment (vabold) [reply include/sdk/heapCommon.hh:13]: These are the only real methods that have any logic, hence why I chose to prioritize them.

**Pull request merged**